pkg/bootstrap/bootstrap.go

```diff
@@ -135,22 +160,30 @@ func extract(image, targetDir, prefix string, reader io.Reader) error {
 	}
 }
 
-func releaseName(ref name.Reference) string {
-	if t, ok := ref.(*name.Tag); ok && releasePattern.MatchString(t.TagStr()) {
+// releaseRefDigest returns a unique name for an image reference.
+// If the image refers to a tag that appears to be a version string, it returns the tag + the first 12 bytes of the SHA256 hash of the reference string.
+// If the image refers to a digest, it returns the digest, without the alg prefix ("sha256:", etc).
+// If neither of the above conditions are met (semver tag or digest), an error is raised.
+func releaseRefDigest(ref name.Reference) (string, error) {
+	if t, ok := ref.(name.Tag); ok && releasePattern.MatchString(t.TagStr()) {
 		hash := sha256.Sum256([]byte(ref.String()))
-		return t.TagStr() + "-" + hex.EncodeToString(hash[:])[:12]
-	} else if d, ok := ref.(*name.Digest); ok {
+		return t.TagStr() + "-" + hex.EncodeToString(hash[:])[:12], nil
+	} else if d, ok := ref.(name.Digest); ok {
 		str := d.DigestStr()
 		parts := strings.SplitN(str, ":", 2)
 		if len(parts) == 2 {
-			return parts[1]
+			return parts[1], nil
 		}
-		return parts[0]
+		return parts[0], nil
 	}
-	return ""
+	return "", fmt.Errorf("Bootstrap image %q is not a not a reference to a digest or version tag (%q)", ref, releasePattern)
```

Comment:
It looks like the runtime image is always either rke2-runtime:v1.18-xxxx or rke2-runtime@sha256:yyyy, so we are safe considering anything else an error... now that we've fixed the type assertions.